app/src/main/java/com/raulh82vlc/functionalkotlinarrow/presentation/Presentation.kt

```diff
@@ -21,29 +21,26 @@ import arrow.effects.IO
 import arrow.effects.fix
 import arrow.effects.monadError
 import arrow.typeclasses.bindingCatch
-import com.raulh82vlc.functionalkotlinarrow.data.cache.model.FeedItemCacheModel
-import com.raulh82vlc.functionalkotlinarrow.domain.FeedUseCase
+import com.raulh82vlc.functionalkotlinarrow.data.cache.model.FeedItemCache
+import com.raulh82vlc.functionalkotlinarrow.domain.getFeed
 
 interface FeedListView {
     fun showNetworkError(t: Throwable)
-    fun showFeed(feed: List<FeedItemCacheModel>)
+    fun showFeed(feed: List<FeedItemCache>)
 }
 
-object Presentation {
-    fun onClickFeedItem(ctx: Context, heroId: String): IO<Unit> =
-            Navigation.goToFeedDetail(ctx, heroId)
+fun onClickFeedItem(context: Context, heroId: String): IO<Unit> = goToFeedDetail(context, heroId)
 
-    fun showFeed(view: FeedListView): IO<Unit> {
-        val monadError = IO.monadError()
-        return monadError.bindingCatch {
-            val result = FeedUseCase.getFeed().handleError {
-                displayError(view, it); emptyList()
+fun showFeed(view: FeedListView): IO<Unit> {
+    val monadError = IO.monadError()
+    return monadError.bindingCatch {
+        monadError.just(
+                view.showFeed(getFeed().handleError {
+                    displayError(view, it); emptyList()
             }.bind()
+        )).bind()
+    }.fix()
+}
```

Comment:
This function feels like needs some formatting. Maybe something like this:
```
fun FeedListView.showFeedIO(): IO<Unit> = IO.monadError().bindingCatch {
    just(showFeed(
        feed = getFeed().handleError {
            showNetworkError(it)
            emptyList()
        }.bind()
    )).bind()
}.fix()
```
However, instead of having a side effect when handling the error it may be worth to return a `IO<Either<List<FeedItemCacheModel>, Error>>` or ``IO<Try<List<FeedItemCacheModel>>>`` if using Throwable instead of own error types